### PR TITLE
Filter out sessions when specific endpoint is hit

### PR DESCRIPTION
**Pull request by tavy88**
### Update

Visit `/status_check` and you should not get a session.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -24,4 +26,8 @@ def track_event(event_key, properties = {})
   def protect_feature(feature)
     redirect_to task_list_path unless Flipflop.enabled?(feature)
   end
+
+  def set_request_path
+    user_session.request_path = request.path
+  end
```
Comment (smoothcontract) [inline app/controllers/application_controller.rb]: I'm not very keen on this approach. This means that every request (apart from the health endpoint) to the site is going to require a database write to update this key on the session. That feels like a bad idea when trying to build a scalable site. We should only be updating the session content where necessary and when the user undertakes a specific action we care about.

File: app/models/session.rb [hunk 2]
```diff
@@ -1,3 +1,13 @@
 class Session < RestrictedActiveRecordBase
   has_one :user
+
+  validate :health_checkpoint
+
+  private
+
+  def health_checkpoint
+    return if data.present? && data['request_path'].exclude?('/status_check')
```
Comment (smoothcontract) [inline app/models/session.rb]: Avoiding saving the session when it's empty is an interesting idea; there may be some mileage in that but would be good if we can find an alternate technique to raising validation errors.

Comment (smoothcontract) [reply app/models/session.rb]: Perhaps look at overriding `save` method or adding a `before_save` callback instead.. 🤔

File: app/controllers/status_check_controller.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class StatusCheckController < ApplicationController
+  def index
+    request.session_options[:skip] = true
```
Comment (cwrw) [inline app/controllers/status_check_controller.rb]: are you sure this works? I checked the issues on active record session and it seems that this option was removed?

Comment (cwrw) [reply app/controllers/status_check_controller.rb]: https://github.com/rails/activerecord-session_store/issues/122

Comment (tavy88) [reply app/controllers/status_check_controller.rb]: I've tested it yes. Feel free to test as well.

**Comment by tavy88** [comment]
This is no longer needed.

**Pull request closed without merging**